File: sc/src/main/scala/Solution_387.scala
class Solution_387 {
  def firstUniqChar(s: String): Int = {
    val charsCounts = Array.fill[Int](26)(0)
    for (ch <- s) {
      charsCounts(ch - 'a') += 1
    }
    for (i <- 0 until s.length - 1) {
      if (charsCounts(s(i) - 'a') == 1) return i
    }
    -1
  }
}

File: kt/src/main/kotlin/id/algo/Solution_387.kt
package id.algo

class Solution_387 {
    fun firstUniqChar(s: String): Int {
        val charCounts = IntArray(26)
        for (char in s) {
            charCounts[char - 'a']++
        }
        for (i in 0..s.length - 1) {
            if (charCounts[s[i] - 'a'] == 1) return i
        }
        return -1
    }
}